algorithm/.gitignore

```diff
@@ -0,0 +1,2 @@
+/target
+.DS_Store
```

Comment:
Please add Cargo.lock. A binary (application) crate should include Cargo.lock in the repository, while a library crate (one without main.rs) should not.